atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/futures/AtlasFutures.java

```diff
@@ -92,4 +96,12 @@ public void close() {
             throw Throwables.rewrapAndThrowUncheckedException(e);
         }
     }
+
+    public static Executor traceRestoringExecutor(Executor executorService, String operation) {
```

Comment:
private

Reply:
Maybe currentTraceExecutor? or a comment explaining how the heck this works, because this is super subtle.

Reply:
It is also used in the CqlClientImpl, unless we want to remove that.

Reply:
Discussed offline: we don't care about it so much in CqlClientImpl, since that stuff is happening directly on the netty threads, it's unlikely we'll do any logging there.